lib/internal/bootstrap_node.js

```diff
@@ -257,6 +258,22 @@
     });
   }
 
+  function setupInspectorAPI() {
+    const addCommandLineAPIMethod = process.inspector.addCommandLineAPIMethod;
+    delete process.inspector.addCommandLineAPIMethod;
+    addCommandLineAPIMethod('require', function require(request) {
+      const Module = NativeModule.require('module');
+      const path = NativeModule.require('path');
+      const cwd = tryGetCwd(path);
+
+      const name = '[consoleAPI]';
+      const module = new Module(name);
+      module.filename = path.join(cwd, name);
+      module.paths = Module._nodeModulePaths(cwd);
+      return module.require(request);
+    });
```

Comment:
The module cache is maintained by `Module`, so individual module instances shouldn't matter. It does influence `module.parent` (e.g. two modules that were loaded initially using the global require will not share the same `module.parent`). But that seemed less important.

Adding a test case to verify.